src/math/CurveFitting.cc

```diff
@@ -477,13 +472,11 @@ namespace pfs{ namespace drp{ namespace stella{ namespace math{
       if (P_D_A1_Sigma->getShape()[0] != nCoeffs){
         string message("pfs::drp::stella::math::CurveFitting::PolyFit: ERROR: P_D_A1_Sigma->getShape()[0](=");
         message += to_string(P_D_A1_Sigma->getShape()[0]) +") != nCoeffs(=" + to_string(nCoeffs) + ")";
-        cout << message << endl;
         throw LSST_EXCEPT(pexExcept::Exception, message.c_str());
       }
-      #ifdef __DEBUG_POLYFIT__
-        cout << "pfs::drp::stella::math::CurveFitting::PolyFit: KeyWord_Set(SIGMA): *P_D_A1_Sigma set to " << (*P_D_A1_Sigma) << endl;
-      #endif
+      LOGLS_DEBUG(_log, "KeyWord_Set(SIGMA): *P_D_A1_Sigma set to " << (*P_D_A1_Sigma));
     }
+    P_D_A1_Sigma->deep() = 0.;
```

Comment:
Again, why is it only necessary to initialize this now? Please explain the motivation. If it's a bug fix, can we have a unit test?